lang/csharp/src/apache/codegen/AvroGen.cs

```diff
@@ -201,5 +220,102 @@ static void Usage()
 
             return 0;
         }
+
+        static int GenSchema(List<string> infiles, string outdir, IEnumerable<KeyValuePair<string, string>> namespaceMapping)
+        {
+            try
+            {
+                var sn = new SchemaNames();
+                CodeGen codegen = new CodeGen();
+                var targetNs = new List<string>();
+
+                if (infiles.Count == 1)
+                {
+                    FileAttributes attr = System.IO.File.GetAttributes(infiles.First());
+                    if (attr.HasFlag(FileAttributes.Directory))
+                    {
+                        var dirInfo = new DirectoryInfo(infiles.First());
+                        infiles = dirInfo.GetFiles("*.*", SearchOption.TopDirectoryOnly)
+                                         .OrderBy(f => f.Name)
+                                         .Select(f => f.FullName)
+                                         .ToList();
+                    }
+                }
+
+                var toRetry = new List<string>();
+                foreach (var infile in infiles)
+                {
+                    FileAttributes attr = System.IO.File.GetAttributes(infile);
+                    if (attr.HasFlag(FileAttributes.Directory))
+                        continue;
+
+                    Console.WriteLine($"Loading Schema from: [{infile}]");
+                    string text = System.IO.File.ReadAllText(infile);
+
+                    //try
```

Comment:
the "retries" system isn't fully built here. I'm working on this primarily from the C# code gen side of things, not from the `avrogen` CLI side of things. However, I and I'm sure @alfhv too) would like to see this finished before merging.